plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -2102,6 +2102,39 @@ private void testBucketedTable(Session session, HiveStorageFormat storageFormat,
         assertFalse(getQueryRunner().tableExists(session, tableName));
     }
 
+    @Test
+    public void testBucketedTableWithBoundedVarchar()
```

Comment:
Can we parameterise this test for different data types ?
You could try something similar to `BaseIcebergConnectorTest#testSplitPruningFromDataFileStatistics`